doc/model_persistence.rst

```diff
@@ -224,12 +233,14 @@ Security & Maintainability Limitations
 --------------------------------------
 
 :mod:`pickle` (and :mod:`joblib` and :mod:`clouldpickle` by extension), has
-many documented security vulnerabilities and should only be used if the
-artifact, i.e. the pickle-file, is coming from a trusted and verified source.
+many documented security vulnerabilities by design and should only be used if
+the artifact, i.e. the pickle-file, is coming from a trusted and verified
+source. You should never load a pickle file from an untrusted source, similarly
+to how you should never execute code from an untrusted source.
 
 Also note that arbitrary computations can be represented using the `ONNX`
-format, and therefore a sandbox used to serve models using `ONNX` also needs to
-safeguard against computational and memory exploits.
+format, and it is therefore recommended to serve models using `ONNX` in a
+sandboxed environment to safeguard against computational and memory exploits.
```

Comment:
@adrinjalali do you have a good reference for the threat model of loading ONNX files from untrusted origins?

It seems that is has some filesystem access (via External Data Files) but overall it seems quite safe to load and run inference on untrusted ONNX files with onnxruntime (assuming no security bugs in onnxruntime itself).

Reply:
I agree with the assessment. It's much better than loading pickle, and the potential issues can be handled if one limits the resources available to the onnxruntime via sandboxing.